tests/test_ckan_to_frictionless.py

```diff
@@ -169,56 +169,82 @@ def test_unjsonify_all_extra_values(self):
             }]
         }
         out = converter.dataset(indict)
         assert out == exp
 
-        # finally what if license*s* already there ...
+        # Remap everything in licenses
         indict = {
-            'licenses': [{
-                'name': 'odc-pddl'
-            }],
-            'license_id': 'odc-odbl'
+            'license_id': 'cc-by',
+            'license_title': 'Creative Commons Attribution',
+            'license_url': 'http://www.opendefinition.org/licenses/cc-by'
         }
         exp = {
             'licenses': [{
-                'name': 'odc-pddl'
+                'name': 'cc-by',
+                'title': 'Creative Commons Attribution',
+                'path': 'http://www.opendefinition.org/licenses/cc-by'
             }]
         }
         out = converter.dataset(indict)
         assert out == exp
 
     def test_dataset_license_with_licenses_in_extras(self):
+        # If CKAN 'extras' contain a list of 'licenses', it must come
+        # from Frictionless.
         indict = {
             'license_id': 'odc-odbl',
-            # TODO: check package_show in CKAN
-            # 'license_title': 'Open Data Commons Open Database License',
-            # 'license_url': 'http://opendefinition.org/licenses/odc-odbl/'
+            'license_title': 'Open Data Commons Open Database License',
+            'license_url': 'https://opendatacommons.org/licenses/odbl/1-0/index.html',
+            'extras': [
+                {
+                    'key': 'licenses',
+                    'value': json.dumps(
+                        {
+                            'licenses': [
+                                {
+                                    'name': 'cc-by',
+                                    'title': 'Creative Commons Attribution',
+                                    'path': 'http://www.opendefinition.org/licenses/cc-by'
+                                },
+                                {
+                                    'name': 'odc-odbl',
+                                    'title': 'Open Data Commons Open Database License',
+                                    'path': 'https://opendatacommons.org/licenses/odbl/1-0/index.html'
+                                }
+                            ]
+                        }
+                    )
+                }
+            ]
         }
+        # Not expecting duplicates in the output. Take the first license
+        # found and use it first in the list, preserving order (here,
+        # it comes from the root with `license_id`, `license_title` and
+        # `license_url`, not from inside "extras"). If it appears again
+        # in the extras (dicts are equal), don't add it again.
+        # Otherwise, we would have the license 'odc-odbl' twice in this case
         exp = {
-            'licenses': [{
-                'name': 'odc-odbl',
-            }]
+            'licenses': [
```

Comment:
As the comment above says (lines 229 to 234), there's a missing license in the output because it checks for duplicates. Also, the first to show in the list of dictionaries is the one from retrieving `license_id`, `license_title` and `license_url`: otherwise, the license with name `cc-by` should be first because it's first in the list of licenses.